### PR TITLE
framework/st_things: Support Artik H/W Certificate & Key

### PR DESCRIPTION
**Pull request by an4967**
- enable artik h/w certificate & key
- apply TizenRT coding rule

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_sss_manager.c:56: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:70: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:72: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/framework/things_sss_manager.c:78: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:91: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:109: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:115: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:137: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:146: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:VxW)
framework/src/st_things/things_stack/framework/things_sss_manager.c:146: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:152: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:182: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:192: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:196: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:210: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/st_things/things_stack/framework/things_sss_manager.c:260: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/framework/things_sss_manager.c:260: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/framework/things_sss_manager.c:287: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:WxV)
framework/src/st_things/things_stack/framework/things_sss_manager.c:298: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:WxV)

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_sss_manager.c:72: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/framework/things_sss_manager.c:146: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:VxW)
framework/src/st_things/things_stack/framework/things_sss_manager.c:298: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:WxV)

File: framework/src/st_things/things_stack/framework/things_security_manager.c [hunk 1]
```diff
@@ -420,6 +423,57 @@ static int GenerateSvrDb()
 	return res;					// return 0 when failed, 1 otherwise..
 }
 
+static void sm_secure_resource_check(OicUuid_t *device_id)
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/framework/things_security_manager.c]: plz check return type

File: framework/src/st_things/things_stack/framework/things_security_manager.c [hunk 2]
```diff
@@ -497,25 +546,46 @@ int sm_generate_mac_based_device_id(bool is_forced)
...
-	return res;
+	return oc_res;
 }
 
-// To support MAC based UUID
-//
+int sm_generate_device_id(void)
+{
+	int ret = -1;
+#if defined(CONFIG_ARCH_BOARD_ARTIK053)
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/framework/things_security_manager.c]: both support artik053 and 053S .

File: framework/src/st_things/things_stack/framework/things_sss_manager.c [hunk 3]
```diff
@@ -0,0 +1,313 @@
+/****************************************************************************
+ *
+ * Copyright 2016 Samsung Electronics All Rights Reserved.
```
Comment (JeongJunSik) [inline framework/src/st_things/things_stack/framework/things_sss_manager.c]: 2016 ??

**Comment by seinfra** [comment]
Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:680: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
framework/src/st_things/things_stack/framework/things_data_manager.c:687: ERROR: [BRC_M_SMT] else should follow close brace '}'
framework/src/st_things/things_stack/framework/things_security_manager.c:448: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
framework/src/st_things/things_stack/framework/things_security_manager.c:456: ERROR: [BRC_M_SMT] else should follow close brace '}'
framework/src/st_things/things_stack/framework/things_security_manager.c:593: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
framework/src/st_things/things_stack/framework/things_security_manager.c:597: ERROR: [BRC_M_SMT] else should follow close brace '}'
framework/src/st_things/things_stack/framework/things_security_manager.c:820: ERROR: [BRC_M_SMT] that open brace { should be on the previous line
framework/src/st_things/things_stack/framework/things_security_manager.c:832: ERROR: [BRC_M_SMT] else should follow close brace '}'

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Pull request merged**